components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/api/builder/WSO2RegistrationEventPayloadBuilder.java

```diff
@@ -0,0 +1,236 @@
+                .initiatorType(initiatorType)
+                .user(newUser)
+                .organization(organization)
+                .userStore(userStore)
+                .registrationMethods(registrationMethods)
+                .credentialsEnrolled(credentialEnrolled)
+                .build();
+    }
+
+    private void enrichUser(UserStoreManager userStoreManager, String userName, Map<String, Object> properties,
+                            User user)
+            throws IdentityEventException {
+
+        if (properties.containsKey(IdentityEventConstants.EventProperty.USER_CLAIMS)) {
+            Map<String, String> claims = (Map<String, String>) properties.get(IdentityEventConstants.EventProperty
+                    .USER_CLAIMS);
+
+            String userId = claims.get(FrameworkConstants.USER_ID_CLAIM);
+            user.setId(userId);
+            user.setRef(
+                    EventPayloadUtils.constructFullURLWithEndpoint(SCIM2_ENDPOINT) + "/" + user.getId());
+
+            List<UserClaim> userClaims = new ArrayList<>();
+            String emailAddress = claims.get(FrameworkConstants.EMAIL_ADDRESS_CLAIM);
+            String givenName = claims.get(FIRST_NAME_CLAIM_URI);
+            String lastName = claims.get(LAST_NAME_CLAIM_URI);
+
+            try {
+                if (StringUtils.isEmpty(emailAddress)) {
+
+                    emailAddress =
+                            userStoreManager.getUserClaimValue(userName, FrameworkConstants.EMAIL_ADDRESS_CLAIM,
+                                    UserCoreConstants.DEFAULT_PROFILE);
+
+                }
+                if (StringUtils.isEmpty(givenName)) {
+
+                    givenName = userStoreManager.getUserClaimValue(userName, FIRST_NAME_CLAIM_URI,
+                            UserCoreConstants.DEFAULT_PROFILE);
+                }
+
+                if (StringUtils.isEmpty(lastName)) {
+
+                    lastName = userStoreManager.getUserClaimValue(userName, LAST_NAME_CLAIM_URI,
+                            UserCoreConstants.DEFAULT_PROFILE);
+
+                }
+            } catch (UserStoreException e) {
+                throw new IdentityEventException(
+                        "Error while extracting user claims for the user : " + user.getId(), e);
+            }
+
+            UserClaim emailAddressUserClaim = new UserClaim(FrameworkConstants.EMAIL_ADDRESS_CLAIM, emailAddress);
+            UserClaim givenNameUserClaim = new UserClaim(FIRST_NAME_CLAIM_URI, givenName);
+            UserClaim lastNameUserClaim = new UserClaim(LAST_NAME_CLAIM_URI, lastName);
+
+            userClaims.add(emailAddressUserClaim);
+            userClaims.add(givenNameUserClaim);
+            userClaims.add(lastNameUserClaim);
+            user.setClaims(userClaims);
+        } else {
+            enrichUser(userStoreManager, userName, user);
+        }
+    }
+
+    private List<String> getRegistrationMethods(String[] internalRoles) {
```

Comment:
This is not the intention of the use of registration method. The intention here is similar to authentication methods like from which authentication method they authenticated. This is similar, and this should rather indicate if it's username/pwd, google, email link etc

Reply:
Given that we don't have context information for this, let's not have this in first cut